libs/ipc/inc/public/tfc/ipc.hpp

```diff
@@ -193,6 +193,17 @@ class signal {
     return signal_->async_send(value, std::forward<completion_token_t>(token));
   }
 
+  template <typename logger_t>
+  auto async_send(logger_t& logger) -> std::function<void(value_t)> {
+    return [this, &logger](value_t value) {
+      signal_->async_send(value, [this, &logger](std::error_code const& send_error, size_t) {
+        if (send_error) {
+          logger.error("{}: Failed to send: {}", name(), send_error.message());
+        }
+      });
+    };
+  }
+
```

Comment:
This is not a good idea.
You are taking reference to an externally owned logger where you have no control over lifetime.

Reply:
I misread this whole thing.